Malnourished

Tetsukiba opened the pull request:
Adds **penalties** for being malnourished

- Lack of nutrition causes liver and stomach damage
- Lack of hydration causes liver and kidney damage
- These organs can begin to fail after 1-2 days in real time
- These organs will be destroyed after 2-3 days in real time

biscuitWizard committed:
Adding persistence modpack.

biscuitWizard committed:
Adjusting gas errors.

biscuitWizard committed:
Fixing serializer bugs.

biscuitWizard committed:
adding admin verbs to persistence modpack

biscuitWizard committed:
Pull App configuration for automatic PRs with upstream.

Andrew-Fall committed:
Merge branch 'dev' into admin_verbs

biscuitWizard committed:
fixed new_player.dm overrides

biscuitWizard committed:
fixing travis errors

biscuitWizard committed:
Merge pull request #12 from PersistentSS13/fix_new_player

fixed new_player.dm overrides

biscuitWizard committed:
Merge branch 'dev' into admin_verbs

biscuitWizard committed:
Merge branch 'dev' into serializer_fixes

Andrew-Fall committed:
Merge remote-tracking branch 'NebulaSS13/dev' into dev

Andrew-Fall committed:
Merge pull request #16 from Andrew-Fall/dev

Update to latest Nebula

Andrew-Fall committed:
Merge branch 'dev' into admin_verbs

biscuitWizard committed:
Merge pull request #8 from PersistentSS13/admin_verbs

adding admin verbs to persistence modpack

biscuitWizard committed:
Merge branch 'dev' into serializer_fixes

biscuitWizard committed:
Adding populate parts override hook to atoms subsystem.

biscuitWizard committed:
Suppressing debug crash on atoms_movable.

biscuitWizard committed:
Adding comment for atoms subsystem on why override exists. Downstream use.

biscuitWizard committed:
fixing a not-flag error in ifdef.

biscuitWizard committed:
Fixing yet more serialization errors and loose-ends.

biscuitWizard committed:
Merge pull request #7 from PersistentSS13/serializer_fixes

Fixing serializer bugs.

biscuitWizard committed:
Adding lobby music and picture. Added music track.

Andrew-Fall committed:
Merge pull request #26 from PersistentSS13/lobby

Adding lobby music and picture. Added music track.

Andrew-Fall committed:
Merge remote-tracking branch 'NebulaSS13/dev' into dev

biscuitWizard committed:
Merge remote-tracking branch 'upstream/dev' into upstream_merge

# Conflicts:
#	code/modules/hydroponics/seed.dm

biscuitWizard committed:
Merge pull request #30 from PersistentSS13/upstream_merge

Upstream merge

biscuitWizard committed:
Adding cloning, cloning program, cloning machine, and adjusted stack to new life flow.

Commit:
ceiling fix

raise the roof

Andrew-Fall committed:
Merge pull request #33 from N8-Toe/Ceilingfix

Adds Ceiling over kitchen

Andrew-Fall committed:
Merge remote-tracking branch 'PersistentNebula/dev' into dev

Andrew-Fall committed:
Merge remote-tracking branch 'NebulaSS13/dev' into dev

biscuitWizard committed:
Merge pull request #34 from Andrew-Fall/dev

Latest Upstream

Commit:
Adding sequential GUIDs to id generator.

biscuitWizard committed:
Documentation update. (#31)

* Documentation update.

* Update installation.md

Commit:
Added chargen spawn pods for narrative and in-play character generation.

Andrew-Fall committed:
Merge pull request #37 from PersistentSS13/sequential_guid

Adding sequential GUIDs to id generator.

biscuitWizard committed:
Merge pull request #25 from PersistentSS13/cloning

Adding cloning, cloning program, cloning machine, and adjusted stack …

Andrew-Fall committed:
Resolve Conflicts

biscuitWizard committed:
Merge pull request #42 from Andrew-Fall/dev

Latest Upstream

pull[bot] committed:
Merge pull request #47 from NebulaSS13/dev

[pull] dev from NebulaSS13:dev

pull[bot] committed:
Merge pull request #49 from NebulaSS13/dev

[pull] dev from NebulaSS13:dev

Commit:
Removed nebula persistence subsystem. Added multi-z and per-area support to saving for persistence.

Commit:
Merge remote-tracking branch 'upstream/dev' into neb_pr

biscuitWizard committed:
Overhauling R&D. (#48)

* Overhauling R&D.

* Converted R&D to use networks.

* Overhauling R&D Part 2 Boogaloo

Co-authored-by: bitMuse <ddffdsa@afdasda.com>

biscuitWizard committed:
Changing stasis values for beds, sleeping, and making cryopods not log you out. (#40)

biscuitWizard committed:
Merge pull request #52 from PersistentSS13/neb_pr

Updating to latest Nebula

Commit:
Merge remote-tracking branch 'upstream/dev' into neb_pr

Andrew-Fall committed:
Merge pull request #54 from PersistentSS13/neb_pr

Updating Nebula

pull[bot] committed:
Merge pull request #65 from NebulaSS13/dev

[pull] dev from NebulaSS13:dev

Commit:
Many, many, many stabilization fixes and serialization optimizations. Including infamous floor fix.

Commit:
Nebula upstream won't fix these files. We encounter these errors in persistence, and thus they are now fixed.

Commit:
Added hotloading of maps.

NataKilar committed:
Adds cortical chat

biscuitWizard committed:
Merge pull request #67 from NataKilar/cortical-chat

Adds cortical chat

Commit:
Adding stellar anchor.

Commit:
Adding starting vending machines in accordance with issue #57

NataKilar committed:
Adds ship and sector creation

biscuitWizard committed:
Merge pull request #51 from PersistentSS13/overmap_saving

Removed nebula persistence subsystem. Added multi-z and per-area supp…

biscuitWizard committed:
Merge pull request #68 from PersistentSS13/starting_vendors

Adding starting vending machines in accordance with issue #57

Commit:
Merge remote-tracking branch 'upstream/dev' into nebula

# Conflicts:
#	code/controllers/subsystems/initialization/materials.dm
#	code/game/turfs/simulated/walls.dm
#	test/check-paths.sh

Commit:
Updating saved vars.

Commit:
Updating code to latest pull.

Commit:
Merge remote-tracking branch 'upstream/dev' into nebula

Commit:
Updating saved vars.

biscuitWizard committed:
Merge pull request #74 from PersistentSS13/nebula

pull from Nebula

Commit:
shrunk outreach

this is why you don't wash your outpost on warm

Commit:
Fixed Step_x

keep steppin

biscuitWizard committed:
Merge pull request #75 from N8-Toe/outreachmap

shrunk outreach

biscuitWizard committed:
Adding vending machines to outreach outpost. Fixed material decls.

Commit:
update Docker-related files

biscuitWizard committed:
Merge pull request #79 from PersistentSS13/add_vending

Adding vending

Commit:
use utf8mb4 in mariadb

Commit:
update Docker-related files

Commit:
xales, you're dead to me

Commit:
finalize docker configurations

Commit:
Merge branch 'dev' of https://github.com/PersistentSS13/Nebula into loldocker

Commit:
Add port mapping for db

biscuitWizard committed:
Merge pull request #82 from r3valkyrie/loldocker

Updated Docker Files

biscuitWizard committed:
Updating SQL schema.

biscuitWizard committed:
Merge pull request #83 from PersistentSS13/sql_update

Updating SQL schema.

biscuitWizard committed:
Merge remote-tracking branch 'upstream/dev' into upstream_merge

# Conflicts:
#	code/modules/persistence/noticeboards.dm

biscuitWizard committed:
Fixing z_level fuckup. Thanks to Andrew.

biscuitWizard committed:
Adjusted wrappers, combining areas and some other specific code into one implementation.

biscuitWizard committed:
Trimming saved_vars.json.

biscuitWizard committed:
Fixed decals and overlays.

biscuitWizard committed:
Merge remote-tracking branch 'upstream/dev' into upstream_merge

# Conflicts:
#	code/game/objects/structures/window.dm
#	nebula.dme

biscuitWizard committed:
Adjusting saved vars for merge patch.

biscuitWizard committed:
Merge errors.

biscuitWizard committed:
Merge pull request #84 from PersistentSS13/upstream_merge

Upstream merge

biscuitWizard committed:
Merge pull request #85 from PersistentSS13/saving_fixes

Branch for persistence/serialization fixes.

biscuitWizard committed:
Fixed new player joining on existing clients.

biscuitWizard committed:
Merge pull request #80 from NataKilar/stellar-anchor-eye

Adds ship and sector creation

biscuitWizard committed:
Merge pull request #87 from PersistentSS13/new_player_fix

Fixed new player joining on existing clients.

biscuitWizard committed:
Fixed station z-levels being unconnected.

biscuitWizard committed:
Merge remote-tracking branch 'upstream/dev' into upstream_merge

# Conflicts:
#	nebula.dme
#	test/check-paths.sh

biscuitWizard committed:
Merge pull request #88 from PersistentSS13/z_fix

Fixed station z-levels being unconnected.

biscuitWizard committed:
Fixed references to mineable walls.

biscuitWizard committed:
Updating tests.

biscuitWizard committed:
Adjusting saved vars.

biscuitWizard committed:
Merge pull request #90 from PersistentSS13/mines_fix

Fixed references to mineable walls.

biscuitWizard committed:
Merge pull request #89 from PersistentSS13/upstream_merge

Upstream merge

NataKilar committed:
Initial overmap saving

biscuitWizard committed:
Updating dockerfile.

biscuitWizard committed:
Merge pull request #91 from PersistentSS13/docker

Updating dockerfile.

biscuitWizard committed:
Updating docker file v2.

biscuitWizard committed:
Merge pull request #92 from PersistentSS13/docker

Updating docker file v2.

biscuitWizard committed:
Updating docker v3

biscuitWizard committed:
Merge pull request #93 from PersistentSS13/docker

Updating docker v3

biscuitWizard committed:
Slimming down docker image.

biscuitWizard committed:
Merge pull request #94 from PersistentSS13/docker

Slimming down docker image.

NataKilar committed:
Further work on overmap saving

biscuitWizard committed:
Made generators eat fuel 5x as slow, and fuel injectors for fusion 10x as slow.

NataKilar committed:
Simplifies area serialization

biscuitWizard committed:
Merge remote-tracking branch 'upstream/dev' into upstream_merge

# Conflicts:
#	Dockerfile
#	code/game/objects/effects/decals/cleanable.dm
#	docker-compose.yml
#	maps/tradeship/tradeship_define.dm
#	nebula.dme

biscuitWizard committed:
Updated saved_vars.json for merge.

biscuitWizard committed:
Fixed compile errors with merge. Merge complete!

biscuitWizard committed:
Removing ministation test.

biscuitWizard committed:
Fixing gibs on player login without spawn point.

biscuitWizard committed:
Merge pull request #104 from PersistentSS13/fuel_rebalance

Made generators eat fuel 5x as slow, and fuel injectors for fusion 10…

biscuitWizard committed:
Updating map defines in main nebula dme.

biscuitWizard committed:
Merge pull request #105 from PersistentSS13/upstream_merge

Upstream merge

biscuitWizard committed:
Fixing an index out of bounds error.

biscuitWizard committed:
Cleaning up chargen. Polishing the feature to some level of acceptable.

biscuitWizard committed:
Merge pull request #41 from PersistentSS13/chargen

Added chargen spawn pods for narrative and in-play character generation.

biscuitWizard committed:
Merge pull request #96 from NataKilar/persistent-overmap

Overmap Saving

NataKilar committed:
Fixes chargen map

Andrew-Fall committed:
Merge pull request #108 from NataKilar/chargen-map-fix

Fixes chargen map

Andrew-Fall committed:
Merge remote-tracking branch 'NebulaSS13/dev' into dev

NataKilar committed:
Merge pull request #109 from Andrew-Fall/dev

Upstream Merge - Also kills Travis :sob:

biscuitWizard committed:
Added configuration for controlling the autosave system and adding automatic restarts. (#97)

Co-authored-by: Andrew <unknownnewkarma@gmail.com>

NataKilar committed:
Skill Changes (#111)

* Adjusts chargen and skills

* Skills adjustments

Andrew-Fall committed:
Create wiki-sync.yml (#112)

Andrew-Fall committed:
Updates outreach

Andrew-Fall committed:
Tests out the new wiki bot (#113)

Andrew-Fall committed:
Update wiki-sync.yml

Hate doing this but gotta fix

Andrew-Fall committed:
Update wiki-sync.yml

Andrew-Fall committed:
Update wiki-sync.yml

Andrew-Fall committed:
Merge remote-tracking branch 'NebulaSS13/dev' into From_Nebula

Andrew-Fall committed:
Merge pull request #116 from Andrew-Fall/From_Nebula

From nebula

NataKilar committed:
Merge pull request #114 from Andrew-Fall/general_updates

Updates outreach and fixes Aghost

NataKilar committed:
Fixes various mob saving bugs (#115)

* Fixes HUD saving

* Additional saved vars

* Better fix for skill saving issues

NataKilar committed:
Adds Asteroid Magnet (#110)

* Adds Asteroid Magnet

* Fixes a couple quality checks

* Indentation fix

* Clarifies some code and comments

Andrew-Fall committed:
Merge remote-tracking branch 'NebulaSS13/dev' into From_Nebula

Andrew-Fall committed:
Compatibility fix

Andrew-Fall committed:
Merge pull request #119 from Andrew-Fall/From_Nebula

From nebula

NataKilar committed:
Fixes and adjusts RnD (#120)

Crezilin committed:
Add files via upload (#117)

NataKilar committed:
Fixes access and crew record saving (#130)

* Fixes access and crew record saving

* Updates includes

Tetsukiba committed:
Merge pull request #2 from PersistentSS13/dev

Update

Tetsukiba committed:
Create malnourished.dm

genessee596 commented:
Hey, Outreach trialmin here.

If you're gonna make hunger kill, please make it so hunger does not drain at all when someone is in a cryopod. At the moment, hunger drain is very slow (but not stopped entirely) in cryo, as hunger simply causes slow movement. If you're going to make it kill, this would be severely punishing for players who spend a long time in cryo for whatever reason.

TL:DR; Make players in cryo not get any hungrier until they leave cryo again.

Tetsukiba committed:
Update malnourished.dm

adds stasis checks to prevent damage while players are in stasis objects

Tetsukiba commented:
> Hey, Outreach trialmin here.
> 
> If you're gonna make hunger kill, please make it so hunger does not drain at all when someone is in a cryopod. At the moment, hunger drain is very slow (but not stopped entirely) in cryo, as hunger simply causes slow movement. If you're going to make it kill, this would be severely punishing for players who spend a long time in cryo for whatever reason.
> 
> TL:DR; Make players in cryo not get any hungrier until they leave cryo again.

Implemented a change, hunger and thirst won't cause damage while player is in stasis, however hunger and thirst will still deplete.

genessee596 commented:
Good work

Tetsukiba committed:
Update malnourished.dm

movement damage is handled by handle_exertion() which will later be handled by DoMove() to determine if damage should be applied if the player receiving the damage is controlling their own movement.

Andrew-Fall commented on mods/persistence/modules/mob/malnourished.dm:
```diff
@@ -0,0 +1,28 @@
+/mob/living/carbon/human/handle_regular_status_updates()
+	if((!nutrition || !hydration) && (!InStasis()))
+		add_chemical_effect(CE_TOXIN, 1)
+		adjustToxLoss(0.5001)
```
Why does this have the trailing .0001?

Tetsukiba replied:
an attempt to heal damage is constantly made, this .0001 allows slow damage to the liver. If the person is not in stasis they will slowly build up toxin and liver damage, which will take real days to do considerable damage. The heal attempt heals .5 per tick. the .0001 is the difference.

Andrew-Fall commented:
I'll go ahead and create something similar on the latest branch, seeing as I never merged it before rebasing
